armory/baseline_models/pytorch/carla_single_modality_object_detection_frcnn.py

```diff
@@ -30,6 +26,16 @@ def get_art_model(
 
     if weights_path:
         checkpoint = torch.load(weights_path, map_location=DEVICE)
+        assert (
+            "roi_heads.box_predictor.cls_score.bias" in checkpoint
+        ), "invalid checkpoint for current model, layers do no match."
+        assert (
+            model.roi_heads.box_predictor.cls_score.out_features
+            == checkpoint["roi_heads.box_predictor.cls_score.bias"].shape[0]
+        ), (
+            f"provided model checkpoint does not match supplied model_kwargs['num_classes']: "
+            f"{model_kwargs['num_classes']} != {checkpoint['roi_heads.box_predictor.cls_score.bias'].shape[0]}"
+        )
```

Comment:
Is this too strict? Will the underlying model ever change its structure? If so, there does not seem to be a good way to grab the output shape for a checkpoint without knowing the names of its layers.

Reply:
If it changes structure, we can update the file.